README.md

```diff
@@ -4,5 +4,58 @@ DynamoRevit
 Dynamo libraries and ExternalApplication implementation for Revit.
 
 Branches:
-- Revit2014 : Built on Revit2014 API, works only for Revit2014
 - Revit2015 : Built on Revit2015 API, works only for Revit2015
+- Revit2016 : Built on Revit2016 API, works only for Revit2016
+- Revit2017 : Built on Revit2017 API, works only for Revit2017
+
+
+### Building DynamoRevit
+
+#### Background Info
+**DynamoRevit** is a library of Dynamo Nodes as well as a set of utilities for interacting with Revit via the Dynamo visual programming language.
+
+**DynamoRevit** has different branches for different versions of Revit. For example to run **DynamoRevit** on Revit 2016 you want the **DynamoRevit** 2016 branch.
+**DynamoRevit** is also built on top of Dynamo, and in some cases requires matching the version of Dynamo as well. For example, to run **DynamoRevit** on Revit 2016 using Dynamo .91 you want to build The Dynamo .91 Branch, then build the **DynamoRevit** .91 2016 branch.
+
+
+####Steps
+Download Dynamo from http://www.github.com/DyanmoDS/Dynamo
+Download **DynamoRevit** from here.
+Build Dynamo.all.sln
+Build DynamoRevit.2013.sln
+
+if you have a version of revit installed, you can build an installer from Dynamo/Tools/Install/...
+
+if you do not have an installed version of revit, you can create an .addin file that points to the Dynamo application manually:
+in the correct Revit adding location for your application add a text file as follows:
+
+```<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<RevitAddIns>
+<AddIn Type="Application">
+<Name>Dynamo For Revit</Name>
+<Assembly>"C:\YourGithubReposPath\DynamoVersionxxx\Bin\Revit_20xx\DynamoRevitVersionSelector.dll"</Assembly>
```

Comment:
The path looks wrong or obsolete nowadays. Mine looks like this:
`E:\MyGitPath\Dynamo\bin\AnyCPU\Debug\Revit_2017\DynamoRevitVersionSelector.dll`